Frontend/ui-driver/src/Flow.purs

```diff
@@ -407,7 +414,21 @@ updateDriverVersion dbClientVersion dbBundleVersion = do
       pure unit
     else pure unit
   else pure unit
-
+data FCMToken = FCMToken String
+
+checkAndUpdateToken :: Int -> FlowBT String Unit
+checkAndUpdateToken count = do
+    if count > 0 then do
+        let token = getValueToLocalStore FCM_TOKEN
+        when (any (_ == token) ["_failed","null","(null)",""]) do
```

Comment:
Where is masked token check? If it's coming as (["_failed","null","(null)",""])  while fetching from localStorage it's not necessary that it is wrong in DB as well. That's why we need to add this check in maskedDeviceToken which we receive when in profile response. Please refactor this as discussed

Reply:
But In wrong cases it will update with token in db and right cases will we overrided

Reply:
It will update the DB with token in wrong cases and if in db it is not correct then also it will be overrided with correct token

Reply:
No

Reply:
What does Profile response API gives you?

Reply:
For user side there is maskedTokenDevice in profileResponse API but not in driver profile Repsone API

Reply:
in DriverSide it is only available in VerifyTokenResp